pkg/controller/logstash/keystore.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package logstash
+
+import (
+	"hash"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	logstashv1alpha1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/logstash/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/keystore"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/logstash/volume"
+)
+
+const (
+	KeystorePassKey = "LOGSTASH_KEYSTORE_PASS" // #nosec G101
+)
+
+var (
+	keystoreCommand          = "/usr/share/logstash/bin/logstash-keystore"
+	initContainersParameters = keystore.InitContainerParameters{
+		KeystoreCreateCommand:         keystoreCommand + " create",
+		KeystoreAddCommand:            keystoreCommand + ` add "$key" --stdin < "$filename"`,
+		SecureSettingsVolumeMountPath: keystore.SecureSettingsVolumeMountPath,
+		KeystoreVolumePath:            volume.ConfigMountPath,
+		Resources: corev1.ResourceRequirements{
+			Requests: map[corev1.ResourceName]resource.Quantity{
+				corev1.ResourceMemory: resource.MustParse("1Gi"),
+				corev1.ResourceCPU:    resource.MustParse("1000m"),
+			},
+			Limits: map[corev1.ResourceName]resource.Quantity{
+				corev1.ResourceMemory: resource.MustParse("1Gi"),
+				corev1.ResourceCPU:    resource.MustParse("1000m"),
+			},
+		},
+	}
+
+	DefaultKeystorePass = corev1.EnvVar{Name: KeystorePassKey, Value: "changeit"}
```

Comment:
Without the password in env, `logstash-keystore` prompts user to input [y/N] to continue.
So, we need to either give [N] to stdin or set the default password

Reply:
We should be able to use `yes` to send the input to stdin, so you could change the `keystoreCommand` to 

```
keystoreCommand = "yes | /usr/share/logstash/bin/logstash-keystore"
```
to pipe `y` in 

or 
```
keystoreCommand = "yes n | /usr/share/logstash/bin/logstash-keystore"
```

to pipe `n` in. 

I'm not sure it matters a huge amount, but it might be easier to document if we don't set a default that doesn't really do anything

Reply:
oh! I am such a fool :( 
will change the command

Reply:
Not at all! 

And thank you!